### PR TITLE
Don't allow CREATE TABLE with quicklz, if server is built without quicklz.

### PR DESCRIPTION
**Pull request by hlinnaka**
Commit 46b21685f7 added a similar check for Zstandard. This makes QuickLZ
behave the same.

Fixes https://github.com/greenplum-db/gpdb/issues/5107

### CODE REVIEW

File: src/backend/access/common/reloptions_gp.c [hunk 1]
```diff
@@ -964,16 +964,24 @@ validate_and_adjust_options(StdRdOptions *result,
...
+					(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+					 errmsg("QuickLZ library is not supported by this build"),
+					 errhint("Compile with --with-quicklz to use QuickLZ compression.")));
+#endif
+			if (result->compresslevel != 1)
+			{
+				if (validate)
+					ereport(ERROR,
+							(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+							 errmsg("compresslevel=%d is out of range for quicklz (should be 1)",
```
Comment (danielgustafsson) [inline src/backend/access/common/reloptions_gp.c]: s/quicklz/QuickLZ/ for consistency?

Comment (berlin-ab) [inline src/backend/access/common/reloptions_gp.c]: I'm surprised that "zstd" wasn't a constant somewhere. The magic numbers of `19` (for zstd) and `1` (for quicklz) should be `#define`d somewhere.

**Comment by ivannovick** [comment]
This makes TONS of sense, thx!

**Comment by hlinnaka** [comment]
Rebased, and added another HAVE_LIBQUICKLZ check. I didn't notice earlier that there was a second copy of essentially the same code that also needs this.

File: src/backend/access/common/reloptions_gp.c (hunk 1 above)
Comment (hlinnaka) [reply src/backend/access/common/reloptions_gp.c]: Hmm. The Zstandard messages have the same pattern. Yeah, might be good to reword these, but I'l leave that to a separate PR.

Comment (hlinnaka) [reply src/backend/access/common/reloptions_gp.c]: Agreed. I'm not going to try fixing that as part of this PR, though.

File: src/backend/access/common/reloptions_gp.c [hunk 2]
```diff
@@ -1185,13 +1193,19 @@ validateAppendOnlyRelOptions(bool ao,
...
+#ifndef HAVE_LIBQUICKLZ
 			ereport(ERROR,
-					(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
-					 errmsg("compresslevel=%d is out of range for quicklz (should be 1)",
-							complevel)));
+					(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+					 errmsg("QuickLZ library is not supported by this build"),
+					 errhint("Compile with --with-quicklz to use QuickLZ compression.")));
+#endif
+			if (complevel != 1)
```
Comment (berlin-ab) [inline src/backend/access/common/reloptions_gp.c]: More evidence that we should create a #define for this expected compression level number.

**Pull request merged**